src/common/debug.c

```diff
@@ -303,3 +303,26 @@ daos_key2str(daos_key_t *key)
 }
 #endif
 
+/* Format a directory entry suitable for logging.
+ * Take a directory entry (filename) and return something suitable for printing, no not modify the
+ * input itself, either return the input as-is for printing or some metadata about it.
+ * TODO: Check boundary case.
+ */
+char *
+daos_de2str(const char *de)
+{
+	int i;
+
+	if (!de)
+		return "<NULL>";
+
+	for (i = 0; i < NAME_MAX; i++) {
+		if (de[i] == '\0')
+			return (char *)de;
+		if (de[i] == '/')
+			return "<contains slash>";
+		if (!isprint(de[i]) || de[i] == '\'')
+			return "<not printable>";
+	}
+	return "<entry too long>";
```

Comment:
So the rules are
1. If `null` print `<NULL>`
2. If contains a `/` print `<contains slash>`
   Why?
4. If contains a `'` print `<not printable>`
   Why?
6. If too long print `<entry too long>`
7. Otherwise print the name.

Reply:
For non-release builds yes, for release builds just print something else.  I might have missed that part of the PR.

The thinking is that for non-release builds we print `de'<name>'` and can then choose to do post-analysis on the logs to remove them.  Having ' in the logs would cause problems with that.  It's the reason I changed the key2str function above not to print ' as well.